R/libpaths.R

```diff
@@ -191,11 +191,17 @@ renv_libpaths_restore <- function() {
 # 'renv::restore(library = <...>)')
 #
 # https://github.com/rstudio/renv/issues/1544
-renv_libpaths_resolve <- function(library) {
-
-  if (is.null(library))
-    return(renv_libpaths_all())
+renv_libpaths_resolve <- function(library = NULL, project = NULL) {
+  if (!is.null(library)) {
+    return(unique(c(library, .Library)))
+  }
 
-  unique(c(library, .Library))
+  if (!is.null(project)) {
+    proj_lib <- renv_paths_library(project = project)
```

Comment:
I'm not sure if this is the desired behavior. Suppose I do something like:

```
setwd("~/test/project")
renv::install("rlang")
```

And `~/test/project` does happen to have an `renv` library, but I haven't actually activated or loaded that project. IMHO, we should install into the current library paths, not the renv project library. That is, I think loading a project explicitly should be a pre-requisite to having `renv::install()` use the project library.

Reply:
But this PR doesn't change that code — this only affects that behaviour when you explicitly supply a `project`. In that case, I think it's reasonable to say that you want to operate on that project, regardless of what `project` is currently loaded in the current session (if any).

i.e. if you do `renv::install("rlang", project = "another-project")` don't you want to install into use `another-project`s library?

But maybe only installing to that project's library if it exists is wrong — if you say you want to install into a given project, and that project doesn't have a library, it seems like it would be better to error?

Reply:
I think my concern is that `project` is usually going to be non-NULL internally, since we fall back to the current working directory when no project is active. For example:

```
> renv:::renv_project_resolve(NULL)
[1] "/Users/kevin"
```

And we use that pretty commonly when resolving the "active" project, e.g. we do that here:

https://github.com/rstudio/renv/pull/1586/files#diff-76ebee0a54b80584fcf1eff0cdf55ef6f17e47a6d80cb807b5939e8e50cac32cR128

I think my fears would be abated if we distinguished between the case where `project` was explicitly supplied by the user, versus we just used the default / current value.

Reply:
Could we add some tests (with multiple `renv` projects) just to confirm the behavior we want here as well? I guess the matrix here is (A x B) / (loaded x unloaded) / (explicit project / implicit project).